### ISSUE TITLE
Inference parameters for Bedrock titan models not working

### ISSUE THREAD

**Issue by aarora79** [hwchase17/langchain#5713]
### System Info

LangChain version 0.0.190
Python 3.9

### Who can help?

@seanpmorgan @3coins 

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Tried the following to provide the `temperature` and `maxTokenCount` parameters when using the `Bedrock` class for the `amazon.titan-tg1-large` model.

```
import boto3
import botocore
from langchain.chains import LLMChain
from langchain.llms.bedrock import Bedrock
from langchain.prompts import PromptTemplate
from langchain.embeddings import BedrockEmbeddings

prompt = PromptTemplate(
    input_variables=["text"],
    template="{text}",
)

llm = Bedrock(model_id="amazon.titan-tg1-large")
llmchain = LLMChain(llm=llm, prompt=prompt)

llm.model_kwargs = {'temperature': 0.3, "maxTokenCount": 512}

text = "Write a blog explaining Generative AI in ELI5 style."
response = llmchain.run(text=text)
print(f"prompt={text}\n\nresponse={response}")
```

This results in the following exception
```
ValueError: Error raised by bedrock service: An error occurred (ValidationException) when calling the InvokeModel operation: The provided inference configurations are invalid
```

This happens because https://github.com/hwchase17/langchain/blob/d0d89d39efb5f292f72e70973f3b70c4ca095047/langchain/llms/bedrock.py#L20 passes these params as key value pairs rather than putting them in the `textgenerationConfig` structure as the Titan model expects them to be,

The proposed fix is as follows:
```
 def prepare_input(
        cls, provider: str, prompt: str, model_kwargs: Dict[str, Any]
    ) -> Dict[str, Any]:
     input_body = {**model_kwargs}
     if provider == "anthropic" or provider == "ai21":
         input_body["prompt"] = prompt
     elif provider == "amazon":
          input_body = dict()
          input_body["inputText"] = prompt
          input_body["textGenerationConfig] = {**model_kwargs}           
      else:
          input_body["inputText"] = prompt

    if provider == "anthropic" and "max_tokens_to_sample" not in input_body:
        input_body["max_tokens_to_sample"] = 50

    return input_body
``` 

```

### Expected behavior

Support the inference config parameters.

**Comment by 3coins** [comment]
@aarora79 
Thanks for looking into the fix. Do you want to submit a PR for this change? I can help verify the change on my end once you do.

**Comment by aarora79** [comment]
Thank you for fixing this @3coins .

**Comment by ventz** [comment]
Seeing this with embeddings still:

langchain: 0.0.249
openai: 0.27.8

```
raise ValueError(f"Error raised by inference endpoint: {e}")
ValueError: Error raised by inference endpoint: An error occurred (ValidationException) when calling the InvokeModel operation: The provided inference configurations are invalid
```

**Comment by paul-bradbeer-adv** [comment]
I am getting the same issue with langchain 0.0.249

**Comment by paul-bradbeer-adv** [comment]
@3coins @aarora79 Hi both, I was getting this error. So upgraded to langchain 0.0.249 in hope of getting this fix. I'm using langchain with amazon bedrock service and still get the same symptom. If I pass an empty inference modifier dict then it works but I have no clue what parameters are being used in AWS world by default and obv. have no control.

**Comment by paul-bradbeer-adv** [comment]
@3coins @aarora79 @ventz This is now working for me. I am thinking perhaps env issues caching an older version and a restart flushing things through. Or something changed between 0.0.259 (as I used yesterday) and 0.0.260 (working today) which I haven't yet checked for.

**Comment by ventz** [comment]
@paul-bradbeer-adv Interesting - good to know.

It could be a cache, because I am using: `langchain==0.0.256` in one environment and `langchain==0.0.259` in another - both working for non-embeddings.

**Comment by HannaHUp** [comment]
I got same error ValueError: Error raised by inference endpoint: An error occurred (ValidationException) when calling the InvokeModel operation: The provided inference configurations are invalid

in sagemaker.
langchain==0.0.256
Image: Data Science 3.0
Kernel: Python 3
Instance type: ml.t3.medium 2 vCPU + 4 GiB

I'm trying to get embedding from those chunks that splited by RecursiveCharacterTextSplitter 

Here is the code:

bedrock_embeddings = BedrockEmbeddings(client=boto3_bedrock)
loader = DirectoryLoader("./Sources", glob="**/*.txt", loader_cls=TextLoader, silent_errors=True)
documents = loader.load()
text_splitter = RecursiveCharacterTextSplitter(
    chunk_size = 1000,
    chunk_overlap  = 100,
)
docs = text_splitter.split_documents(documents)
vectorstore_faiss = FAISS.from_documents(
    docs,
    bedrock_embeddings,
)

wrapper_store_faiss = VectorStoreIndexWrapper(vectorstore=vectorstore_faiss)

funny thing is if my doc is smaller(docs[:5]), it worked. 
vectorstore_faiss = FAISS.from_documents(
    docs[:5],
    bedrock_embeddings,
)

**Comment by rmartine-ias** [comment]
@HannaHUp do you get the error if you set `chunk_size` to 512? That was what was causing it for us -- any docs under 512 tokens were handled, but longer stuff gave this (super unhelpful) error.

**Comment by HannaHUp** [comment]
@rmartine-ias  I have tried input size 1000+. no error.

**Comment by rmartine-ias** [comment]
What is maybe important is the limit is in tokens -- does using this splitter work?

```python
from langchain.text_splitter import TokenTextSplitter

docs = TokenTextSplitter(
        encoding_name="gpt2",
        chunk_size=512,
        chunk_overlap=100,
    ).split_documents(data)
```

(we tested and titan seems to use the GPT2 embeddings)
